main/src/main/scala/sbt/BackgroundJobService.scala

```diff
@@ -100,4 +100,5 @@ abstract class JobHandle {
   def id: Long
   def humanReadableName: String
   def spawningTask: ScopedKey[_]
+  @volatile var isAutoCancel: Boolean
```

Comment:
This looks out of place to the otherwise nice interface. Could we make this a `def` like others, and then we can pattern match on `JobHandle` to see if it's a `ThreadJobHandle`, and then access the var?